ios/CodePush/CodePush.m

```diff
@@ -61,7 +62,11 @@ @implementation CodePush {
 
 + (NSURL *)binaryBundleURL
 {
-    return [[NSBundle mainBundle] URLForResource:bundleResourceName withExtension:bundleResourceExtension];
+    NSBundle *bundle = [NSBundle mainBundle];
+    if (isAppExtension && [[bundle.bundleURL pathExtension] isEqualToString:@"appex"]) {
```

Comment:
I totally agree, but there is the case where we want to have a separate bundle for your app extension...
Maybe we could juste have a `bundle` variable, default to `[NSBundle mainBundle]` and edit this variable in extension's source. Or juste expose isAppExtension...